source/extensions/retry/host/omit_host_metadata/omit_host_metadata.cc

```diff
@@ -0,0 +1,20 @@
+#include "extensions/retry/host/omit_host_metadata/omit_host_metadata.h"
+
+#include "common/config/metadata.h"
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+bool OmitHostsRetryPredicate::shouldSelectAnotherHost(const Upstream::Host& host) {
+  return labelSet_.size() &&
```

Comment:
I'd prefer the explicitness of `empty()` instead of `size()` here. A comment would also be nice since its not clear why this check is done

Reply:
Done.